File: app/src/main/java/com/innovatube/boilerplate/injection/component/ActivityComponent.java
package com.innovatube.boilerplate.injection.component;



import com.innovatube.boilerplate.injection.PerActivity;
import com.innovatube.boilerplate.injection.module.ActivityModule;
import com.innovatube.boilerplate.ui.signup.CreateAccountActivity;

import dagger.Component;

/**
 * Created by TOIDV on 4/4/2016.
 */

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)


public interface ActivityComponent {

    void inject(CreateAccountActivity createAccountActivity);


}

File: app/src/main/java/com/innovatube/boilerplate/util/di/components/ActivityComponent.kt
package com.innovatube.boilerplate.util.di.components

import com.innovatube.boilerplate.presentation.main.MainActivity
import com.innovatube.boilerplate.util.di.modules.ActivityModule
import com.innovatube.boilerplate.util.di.modules.FragmentModule
import com.innovatube.boilerplate.util.di.scopes.ActivityScope
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = [(ActivityModule::class)])
interface ActivityComponent {
    fun plus(module: FragmentModule): FragmentComponent
    fun inject(mainActivity: MainActivity)
}
